boba-chain-ops/cmd/boba-regenerate/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ledgerwatch/log/v3"
+	"github.com/urfave/cli/v2"
+
+	"github.com/bobanetwork/v3-anchorage/boba-chain-ops/genesis"
+)
+
+func main() {
+	log.Root().SetHandler(log.StreamHandler(os.Stderr, log.TerminalFormat()))
+
+	app := &cli.App{
+		Name:  "boba-regenerate",
+		Usage: "Use engine api to regenerate the chain",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "l2-private-endpoint",
+				Value:   "http://localhost:8551",
+				Usage:   "Private endpoint for the L2 node",
+				EnvVars: []string{"L2_PRIVATE_ENDPOINT"},
+			},
+			&cli.StringFlag{
+				Name:    "l2-public-endpoint",
+				Value:   "http://localhost:9545",
+				Usage:   "Public endpoint for the L2 node",
+				EnvVars: []string{"L2_PUBLIC_ENDPOINT"},
+			},
+			&cli.StringFlag{
+				Name:    "l2-legacy-endpoint",
+				Value:   "http://localhost:8545",
+				Usage:   "Legacy endpoint for the L2 node",
+				EnvVars: []string{"L2_LEGACY_ENDPOINT"},
+			},
+			&cli.StringFlag{
+				Name:    "jwt-secret-path",
+				Usage:   "Path to the file containing the JWT secret",
+				EnvVars: []string{"JWT_SECRET_PATH"},
+			},
+			&cli.StringFlag{
+				Name:    "rpc-time-out",
+				Usage:   "Timeout for the RPC requests",
+				Value:   "10s",
+				EnvVars: []string{"RPC_TIME_OUT"},
+			},
+			&cli.StringFlag{
+				Name:    "polling-interval",
+				Usage:   "Interval between sending a request to the L2 node to build a new block",
+				Value:   "100ms",
+				EnvVars: []string{"POLLING_INTERVAL"},
+			},
+			&cli.Int64Flag{
+				Name:    "hard-fork-block-number",
+				Usage:   "Block number at which the hard fork will happen",
+				Value:   0,
```

Comment:
Do we want to give this a default, or require the user to set it?

Reply:
If value is 0, then it will pick the latest block as the hard fork block number.

Reply:
Nothing strictly wrong with that, just know that it would never be used this way in production, so wasn't sure if we wanted to explicitly require a `0` be set for 'latest' behavior, instead of implicitly getting it set.  If you're happy with the current behavior, no need to change, just wanted to raise it for discussion.